functions/brinks/src/lib/stripe/validateCheckoutSession.js

```diff
@@ -0,0 +1,68 @@
+import debug from 'debug';
+import * as Sentry from '@sentry/node';
+import { dataSources } from '@thatconference/api';
+import { Firestore } from '@google-cloud/firestore';
+
+const dlog = debug('that:api:functions:brinks:validateCheckoutSession');
+const memberStore = dataSources.cloudFirestore.member;
+const productStore = dataSources.cloudFirestore.product;
+const firestore = new Firestore();
+
+export default function validateCheckoutSession(checkoutSession) {
+  dlog('validateCheckoutSession called id: %s', checkoutSession.id);
+
+  const checkoutLineItems = checkoutSession.line_items.data;
+  const { metadata } = checkoutSession;
+  const { memberId, eventId, productIds: rawProductIds } = metadata;
+  const productIds = JSON.parse(rawProductIds);
+
+  const memberFunc = memberStore(firestore).get(memberId);
+  const productFunc = productStore(firestore).getBatch(productIds);
+
+  return Promise.all([memberFunc, productFunc]).then(result => {
+    const [member, products] = result;
+    const errorList = [];
+    if (member.stripeCustomerId !== checkoutSession.customer) {
+      errorList.push(`member.stripeCustomerId doesn't match stripeCustomerId`);
+    }
+    if (products.length !== productIds.length) {
+      errorList.push(`Not all Products found in product collection`);
+    }
+    if (
+      products.filter(p => p.eventId === eventId).length !== productIds.length
+    ) {
+      errorList.push(
+        `Product(s) in order which doesn't exist in eventId ${eventId}`,
+      );
+    }
+    const matchedLineItems = checkoutLineItems.filter(lineItem =>
+      products.some(p => p.processor.itemRefId === lineItem.price.id),
+    );
+    if (matchedLineItems.length !== checkoutLineItems.length) {
+      errorList.push(
+        `Checkout Session line items match on products not equal. LineItems: ${checkoutLineItems.length}, Matched: ${matchedLineItems.length}`,
+      );
+    }
+
+    // TODO: check if product is currently on sale?
```

Comment:
Is this still valid?

Reply:
Yes, as a decision needs to be made if we want to verify sale dates during this verification step. Currently we are not and I believe it will be more clear on the requirement at this point as we move through more of this process.